specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/AppSpaceResources.json

```diff
@@ -0,0 +1,463 @@
+            "required": true,
+            "description": "Properties of an App Space Resource that need to be updated",
+            "schema": {
+              "$ref": "#/definitions/AppSpaceResourcePatch"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok",
+            "schema": {
+              "$ref": "#/definitions/AppSpaceResource"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Patch AppSpaceResource": {
+            "$ref": "./examples/AppSpaceResources_Patch.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DeploymentInformation": {
+      "description": "App Space Resource Deployment Information - used to track the deployment information for an App Space Resource's tracked ARM resource",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "ARM Id of the deployment",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Provisioning State for the deployment",
+          "type": "string"
+        }
+      }
+    },
+    "RepositoryInformation": {
+      "description": "App Space Resource Repository Information - used to track the github repository for an App Space Resource's tracked ARM resource",
+      "type": "object",
+      "properties": {
+        "url": {
+          "description": "url for the repository",
+          "type": "string"
+        },
+        "branch": {
+          "description": "branch in the repository",
+          "type": "string"
+        }
+      }
+    },
+    "AppSpaceResourceProperties": {
+      "description": "AppSpaceResource resource-specific properties",
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "ARM resource Id of the tracked resource.",
+          "type": "string"
+        },
+        "friendlyName": {
+          "description": "Friendly name of the App Space Resource.",
+          "type": "string"
+        },
+        "resourceLocation": {
+          "description": "Location of the tracked ARM resource. Can be different from the location of the app space resource.",
+          "type": "string"
+        },
+        "repo": {
+          "$ref": "#/definitions/RepositoryInformation",
+          "description": "Optional App Space Resource Repository Information"
+        },
+        "deployment": {
+          "$ref": "#/definitions/DeploymentInformation",
+          "description": "Optional App Space Resource Deployment Information"
+        },
+        "isFavorite": {
```

Comment:
>isFavorite [](http://example.com/codeflow?start=9&length=10)

ARM recommends enum instead of boolean, even if there are only two values. In this case, you could use something like appSpaceLabel: { Favorite, Default } or hightlightState: { Favorite, <others> } , or something similar.

Reply:
changed to enum

Reply:
Might also consider a more general name for the property itself, since the enum value now indicates it's a favorite. That's why I suggested appSpaceLabel as the property name: it allows for other label values in the future.

Reply:
changed enum name to appSpaceResourceLabel per comment.